jupyter_telemetry/eventlog.py

```diff
@@ -167,7 +203,25 @@ def record_event(self, schema_name, version, event, timestamp_override=None):
             '__timestamp__': timestamp.isoformat() + 'Z',
             '__schema__': schema_name,
             '__schema_version__': version,
-            '__metadata_version__': TELEMETRY_METADATA_VERSION
+            '__metadata_version__': TELEMETRY_METADATA_VERSION,
         }
-        capsule.update(event)
+
+        # Filter properties in the incoming event based on the
+        # allowed categories and properties from the eventlog config.
+        allowed_categories = self.get_allowed_categories(schema_name)
+        allowed_properties = self.get_allowed_properties(schema_name)
+
+        # Iterate through the event properties, and only record the
+        # properties labelled with allowed_categories
+        for property_name, data in event.items():
+            prop_categories = schema["properties"][property_name]["categories"]
+            # If the property is explicitly listed in
+            # the allowed_properties, then include it in the capsule
```

Comment:
Instead of excluding it from the capsule, set it to null to indicate missing value?